src/term/display.rs

```diff
@@ -94,21 +96,14 @@ impl Tag {
   }
 }
 
-impl Display for Pattern {
+impl fmt::Display for Pattern {
   fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
     match self {
       Pattern::Var(None) => write!(f, "*"),
       Pattern::Var(Some(nam)) => write!(f, "{nam}"),
-      Pattern::Ctr(nam, pats) => {
-        write!(f, "({}{})", nam, DisplayJoin(|| pats.iter().map(|p| display!(" {p}")), ""))
-      }
+      Pattern::Ctr(nam, pats) => write!(f, "({}{})", nam, pats.iter().map(|p| format!(" {p}")).join("")),
```

Comment:
We have DisplayJoin for displaying joins of iterators, which avoids copying everything